src/traces/contour/plot.js

```diff
@@ -423,7 +423,7 @@ function makeLinesAndLabels(plotgroup, pathinfo, gd, cd0, contours, perimeter) {
 
                 var isOpen = d3.select(this).classed('openline');
 
-                if(pathLen < textWidth * constants.LABELMIN) return;
+                if(pathLen < (textWidth + textHeight) * constants.LABELMIN) return;
```

Comment:
I didn't increase `LABELMIN`, instead included both width and height of the label. That way when you have long labels you'll still get them on about the same length paths as before, but short labels will require a bit longer paths than they did previously:
![screen shot 2017-06-26 at 2 34 00 pm](https://user-images.githubusercontent.com/2678795/27554503-9c74307c-5a7c-11e7-929a-48813e0ab9e5.png)

Note that this isn't in the cost function, because that controls label location on the path but only weakly affects whether the path gets a label at all - ie if it pushes past `MAXCOST`